docs/dev/guidelines/client-design.rst

```diff
@@ -0,0 +1,241 @@
+        |
+        | Keep this in mind while you select the colors to use.
+
+    Let's go through it step by step. Let's say, you want to give a box a special background color.
+
+    1. Define a class for it in your component's SCSS file, and use a new, unique variable name as value:
+
+        .. code-block:: scss
+
+            .my-box {
+                background-color: var(--my-special-component-my-box-background-color);
+            }
+
+    2. Add the variable as SCSS variable to both theme variable sheets (``_default-variables.scss`` and ``_dark-variables.scss``):
+
+        .. code-block:: scss
+
+            // My Special Component
+            $my-special-component-my-box-background-color: [...];
+
+    3. Select a value for each theme.
+
+        | **Re-use preset colors where possible!** For example, choose ``gray-700`` in light mode and ``gray-400`` in dark mode.
+        | This is still a good approach as you-re reusing already provided colors.
+
+        .. code-block:: scss
+
+            // My Special Component
+            $my-special-component-my-box-background-color: $gray-600;
+
+        .. IMPORTANT::
+            The two options below are meant as a fallback for special cases.
+            Please justify the use of these options in your PR description.
+
+        If you need a **slightly modified preset color**, use `SCSS functions <https://sass-lang.com/documentation/modules/color>`_ to derive it.
+
+        .. code-block:: scss
+
+            // My Special Component
+            $my-special-component-my-box-background-color: darken($success, 20);
+
+        **If all of this doesn't work, define your own colors.**
+
+        .. code-block:: scss
+
+            // My Special Component
+            $my-special-component-my-box-background-color: #123456;
+
+1. Theme-specific global styles
+===============================
+
+It might happen that you need to modify a global style rule in one of the themes, for example if you're using a external library which styles need to be overridden.
+
+| Each theme has a dedicated file custom global styles can be added to: ``theme-dark.scss`` and ``theme-default.scss``.
+| For styles that should be applied in both themes, use ``global.scss``.
+
+2. Theme Service
+================
+
+There will be occasions where you need to know in your components which theme is currently applied.
+The ``ThemeService`` will provide this information where needed.
+For example, you could add a reactive flag to your component that indicates whether or not the current theme is dark:
+
+.. code-block:: ts
+
+    @Component({
+        selector: 'jhi-my-component',
+        templateUrl: './my-component.component.html',
+        styleUrls: ['my-component.component.scss'],
+    })
+    export class MyComponent implements OnInit, OnDestroy {
+        isDark = false;
+        themeSubscription: Subscription;
+
```

Comment:
Is a constructor and dependency injection necessary to use the themeService?
If yes, I would add it here

Reply:
Yup, forgot it, you're right 👍